src/o365/spo/commands/web/web-clientsidewebpart-list.ts

```diff
@@ -0,0 +1,139 @@
+import auth from '../../SpoAuth';
+import config from '../../../../config';
+import commands from '../../commands';
+import * as request from 'request-promise-native';
+import GlobalOptions from '../../../../GlobalOptions';
+import {CommandOption,
+  CommandValidate, CommandError
+} from '../../../../Command';
+import SpoCommand from '../../SpoCommand';
+import Utils from '../../../../Utils';
+import { GetClientSideWebPartsRsp } from './GetClientSideWebPartsRsp';
+const vorpal: Vorpal = require('../../../../vorpal-init');
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  webUrl: string;
+}
+
+class SpoWebClientSideWebPart extends SpoCommand {
+  
+  public get name(): string {
+    return commands.WEB_CLIENTSIDEWEBPART_LIST;
+  }
+
+  public get description(): string {
+    return 'Lists available client-side web parts';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    
+    return telemetryProps;
+  }
+
+  public commandAction(cmd: CommandInstance, args: CommandArgs, cb: () => void): void {
+    auth
+    .getAccessToken(auth.service.resource, auth.service.refreshToken as string, cmd, this.debug)
```

Comment:
This causes the command to fail if you are connected to the tenant admin (`https://contoso-admin.sharepoint.com`) and try to get info about web part for a regular site, like `https://contoso.sharepoint.com/sites/team-a`. In this case, the above code will retrieve access token for the tenant admin site, instead of the site specified in the webUrl option. Instead the code above should retrieve the resource from the webUrl like:

```ts
const resource: string = Auth.getResourceFromUrl(args.options.webUrl);
auth
  .getAccessToken(resource, auth.service.refreshToken as string, cmd, this.debug)
```